warehouse/query-core/src/main/java/datawave/query/transformer/GroupingTransform.java

```diff
@@ -93,81 +163,185 @@ private Key createAttrKey(String fieldName) {
+     * [MALE, 20],
+     * [40, MALE],
+     * [40, MALE],
+     * [MALE, 22] x 2,
+     * [FEMALE, 18],
+     * [MALE, 24],
+     * [20, MALE],
+     * [30, MALE],
+     * [FEMALE, 18],
+     * [34, MALE]]
+     * 
+     * to create documents list: [
+     * {AGE=16, COUNT=1, GENDER=MALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {COUNT=1, ETA=20, GENERE=MALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {COUNT=1, ETA=40, GENERE=MALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {AGE=40, COUNT=1, GENDER=MALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {COUNT=2, ETA=22, GENERE=MALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {AGE=18, COUNT=1, GENDER=FEMALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {COUNT=1, ETA=24, GENERE=MALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {AGE=20, COUNT=1, GENDER=MALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {AGE=30, COUNT=1, GENDER=MALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {COUNT=1, ETA=18, GENERE=FEMALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false,
+     * {AGE=34, COUNT=1, GENDER=MALE}:20130101_0 test%00;-d5uxna.msizfm.-oxy0iu: [ALL] 1356998400000 false]
+     * 
+     * which is then flattened to just one document with the fields and counts correlated with a grouping context suffix:
+     * 
+     * {
+     * AGE.0=16, GENDER.0=MALE, COUNT.0=1,
+     * ETA.1=20, GENERE.1=MALE, COUNT.1=1,
+     * ETA.2=40, GENERE.2=MALE, COUNT.2=1,
+     * AGE.3=40, GENDER.3=MALE, COUNT.3=1,
+     * ETA.4=22, GENERE.4=MALE, COUNT.4=2,
+     * AGE.5=18, GENDER.5=FEMALE, COUNT.5=1,
+     * ETA.6=24, GENERE.6=MALE, COUNT.6=1,
+     * AGE.7=20, GENDER.7=MALE, COUNT.7=1,
+     * AGE.8=30, GENDER.8=MALE, COUNT.8=1,
+     * ETA.9=18, GENERE.9=FEMALE, COUNT.9=1,
+     * AGE.A=34, GENDER.A=MALE, COUNT.A=1,
+     * }
+     * </pre>
+     *
+     * The Attributes, which have had their visibilities merged, are copied into normal TypeAttributes for serialization to the webserver.
+     *
+     * @param documents
+     */
+    private void flatten(List<Document> documents) {
+        log.trace("flatten {}", documents);
+        Document theDocument = new Document(documents.get(0).getMetadata(), true);
+        int context = 0;
+        Set<ColumnVisibility> visibilities = new HashSet<>();
+        for (Document document : documents) {
+            log.trace("document: {}", document);
+            for (Entry<String,Attribute<? extends Comparable<?>>> entry : document.entrySet()) {
+                String name = entry.getKey();
+                visibilities.add(entry.getValue().getColumnVisibility());
+                
+                Attribute<? extends Comparable<?>> attribute = entry.getValue();
```

Comment:
Do we need to handle Attributes (which extends Attribute) here?  In that case we may have a set of Attribute contained therein.

Reply:
The Attributes in the list of Documents to be flattened are either TypeAttribute (if they are the COUNT attribute I created) or they are GroupingTypeAttribute (if they are the unique combinations to be counted).